jvandijk/com.sensimity.ti.client
# [WIP] Run a beaconscanner inside a started service on android

- [x] Optimize backgroundmode (see https://github.com/Sensimity/android-altbeacon-module/pull/3)
- [x] Run a service the right way, in a started service, separate from the application. It continues iBeacon scans after closing the root-window, by starting it in a background-service.

WIP:
- [x] Update sensimity library to 0.3.0
- [x] Update configuration description and documentation to add a network_id by executing `runService()`.
## Configure
1. Set the following information
   `app/lib/services/handleBackgroundScan.js`

```
var service = Ti.Android.currentService;
var serviceIntent = service.intent;
var sensimity = require('com.sensimity.ti.client');

// Stop sensimity on taskremoved service
service.addEventListener('taskremoved', function(){
    sensimity.stop();
});

sensimity.start({
    networkId: 3,
    runInService: true
});
```

`app/config.json`

```
"backgroundService": "services/handleBackgroundScan.js"
```

`tiapp.xml`

```
<service url="services/handleBackgroundScan.js" type="standard"/>
```

`app/alloy.js`

```
(function () {
    var sensimity = require('com.sensimity.ti.client');
    sensimity.runService(); // Run the sensimityservice inside the background
})();
```
## How to test
1. Start the application
2. Tap back and see the application continues scanning
3. Swipe the application from the application list and see the beaconscanner is unbinding and sensimity stops detecting beacons.
